src/test/java/com/exasol/exasoltestsetup/ExasolTestSetupTestBase.java

```diff
@@ -0,0 +1,188 @@
+            throws InterruptedException, BucketAccessException, TimeoutException, IOException {
+        final Bucket bucket = this.testSetup.getDefaultBucket();
+        bucket.uploadStringContent(TEST_CONTENT, TEST_FILE_NAME);
+        final Path localFile = tempDir.resolve(TEST_FILE_NAME);
+        bucket.downloadFile(TEST_FILE_NAME, localFile);
+        assertThat(Files.readString(localFile), equalTo(TEST_CONTENT));
+    }
+
+    @Test
+    void testMakeLocalServiceAvailableInDatabase() throws Exception {
+        final ExasolTestSetupTestBase.DummySocketServer dummySocketServer = new ExasolTestSetupTestBase.DummySocketServer();
+        try (final ExasolTestSetup testSetup = this.getExasolTestSetup()) {
+            this.statement.executeUpdate("CREATE SCHEMA TEST");
+            final String mappedHostName = testSetup.makeLocalServiceAvailableInDatabase(TEST_SOCKET_PORT);
+            final String pingUdf = "CREATE OR REPLACE PYTHON SCALAR SCRIPT TEST.PING() RETURNS INT AS\n" + //
+                    "def run(ctx):\n" + //
+                    "  import socket\n" + //
+                    "  s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)\n" + //
+                    "  s.connect((\"" + mappedHostName + "\", 5111))\n" + "" + //
+                    "  return 1" + //
+                    "\n/\n";
+            this.statement.executeUpdate(pingUdf);
+            this.statement.executeQuery("select TEST.PING();");
+            Assertions.assertTrue(dummySocketServer.hasClient.get());
+        } finally {
+            dummySocketServer.shutdown();
+        }
+    }
+
+    @Test
+    void testMakeDatabaseServiceAvailableAtLocalhost() throws SQLException {
+        this.statement.executeUpdate("CREATE SCHEMA TEST");
+        final List<Integer> localPorts = this.testSetup.makeDatabaseServiceAvailableAtLocalhost(8001);
+        assertThat(localPorts.size(), greaterThanOrEqualTo(1));
+        final ConnectionTester connectionTester = new ConnectionTester(localPorts);
+        final String pingUdf = "CREATE OR REPLACE PYTHON3 SCALAR SCRIPT TEST.SERVE() RETURNS INT AS\n" + //
+                "def run(ctx):\n" + //
+                "  hadClient = 0\n" + //
+                "  from socketserver import BaseRequestHandler, TCPServer\n" + //
+                "  class handler(BaseRequestHandler):\n" + //
+                "    def handle(self):\n" + //
+                "      hadClient = 1\n" + //
+                "      self.request.sendall(b'hallo')\n" + //
+                "  TCPServer.allow_reuse_address = True\n" + //
+                "  with TCPServer(('', 8001), handler) as server:\n" + //
+                "    server.timeout = 100\n" + //
+                "    server.handle_request()\n" + //
+                "    server.server_close()\n" + //
+                "  return hadClient" + //
+                "\n/\n";
+        this.statement.executeUpdate(pingUdf);
+        this.statement.executeQuery("select TEST.SERVE();");
```

Comment:
Can we maybe reorganize this method as:
- start the server in the database (e.g. extract lines 97 to 113 to a dedicated method)
- make the database server accessible from localhost
- perform ping from localhost
- assert it worked OK